File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.py
class Solution(object):
    def rearrangeArray(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        pos = 0 
        neg = 1 
        ans = [0] * len(nums) 

        for ele in nums :
            if ele > 0 :
                ans[pos] = ele 
                pos += 2 
            else :
                ans[neg] = ele 
                neg += 2 
        return ans         

        

File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        int[] postive = new int[nums.length/2] ; 
        int[] negative = new int[nums.length/2] ;
        int p = 0 ; 
        int n = 0 ; 

        for ( int ele : nums ) {
            if(ele < 0 ) {
                negative[n++] = ele ; 
            }else {
                postive[p++] = ele ; 
            }
        } 
        p=0;
        n=0;
        int i = 0 ; 

        while ( p + n != nums.length ) {
            if(i%2 == 0 ) {
                nums[i++] = postive[p++] ; 
            }
            else {
                nums[i++] = negative[n++] ; 
            }
        }

        return nums ; 
    }
}